File: questions/graphs/bfs.java
class Solution
{
    public ArrayList<Integer> bfsOfGraph(int V,ArrayList<ArrayList<Integer>> adj, boolean visited)
    {
        // Code here
        Queue<Integer> queue = new LinkedList<>();
        queue.add(0);
        while(!q.isEmpty()){
            int front= q.remove();
            if( !visited[front] ){
                result.add(front);
            }
            visited[front] = true;
            //adj.get(Front) --> the list of nodes adj to front
            //adj.get(front).get(i) --> node which is adj to front
            
            for(int i=0; i<adj.get(front).size(); i++){
                int adjNode = adj.get(front).get(i);
                if (!visited[adjNode]){
                    q.add(adjNode);
                }
            }
            return result;
        }
        
    }

    //Function to return Breadth First Traversal of given graph.
    public ArrayList<Integer> bfsOfGraph(int V,ArrayList<ArrayList<Integer>> adj)
    {
        // Code here
        boolean visited[] = new boolean [V];
        for(int i=0; i<V; i++){
            
        }
        
    }
}

File: questions/graphs/bfs.py
#User function Template for python3
import collections

class Solution:
    
    #Function to return Breadth First Traversal of given graph.
    def bfsOfGraph(self, V, adj):
        # code here
        
        q = collections.deque([0])
        visited = [False]*V
        # print(V, adj)
        ans = []
        while q:
            p = q.popleft()
            # print(p)
            if not visited[p]:
                ans.append(p)
                visited[p] = True
                for child in adj[p]:
                    q.append(child)
            # print(q)
            
        # ans = adj
        return ans
        

#{ 
#  Driver Code Starts



if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		V, E = map(int, input().split())
		adj = [[] for i in range(V)]
		for _ in range(E):
			u, v = map(int, input().split())
			adj[u].append(v)
		ob = Solution()
		ans = ob.bfsOfGraph(V, adj)
		for i in range(len(ans)):
		    print(ans[i], end = " ")
		print()
        

# } Driver Code Ends
